checks.d/subdir_sizes.py

```diff
@@ -44,18 +45,41 @@ def check(self, instance):
         subdirtagname = instance.get("subdirtagname", "subdir")
         subdirtagname_regex = instance.get("subdirtagname_regex", "")
         pattern = instance.get("pattern", "*")
+        recurse = instance.get("recurse", False)
 
         if not exists(abs_directory):
             raise Exception("DirectoryCheck: the directory (%s) does not exist" % abs_directory)
 
-        self._get_stats(abs_directory, dirtagname, subdirtagname, subdirtagname_regex, pattern)
+        self._get_stats(abs_directory, dirtagname, subdirtagname, subdirtagname_regex, pattern, recurse)
 
-    def _get_stats(self, directory, dirtagname, subdirtagname, subdirtagname_regex, pattern):
+    def _get_stats(self, directory, dirtagname, subdirtagname, subdirtagname_regex, pattern, recurse):
         orig_dirtags = [dirtagname + ":%s" % directory]
-        directory_bytes = 0
-        directory_files = 0
-        recurse_count = 0
+
+        # Initialize state for subdirectories
+        subdirs = {}
+        for root, dirs, files in walk(directory):
+            if root == directory:
+                for d in dirs:
+                    subdir_path = join(root, d)
+                    if subdirtagname_regex:
+                        pat = re.compile(subdirtagname_regex)
```

Comment:
Can we move this `compile` outside of the loop? I'm assuming that pre-compiling a regex is faster and since DD checks are time sensitive, we should utilize that if we can!